src/Avalonia.Base/Data/BindingOperations.cs

```diff
@@ -43,8 +44,12 @@ public static class BindingOperations
             {
                 case BindingMode.Default:
                 case BindingMode.OneWay:
-                    return target.Bind(property, binding.Observable ?? binding.Subject, binding.Priority);
+                    if (binding.Observable is null)
+                        throw new InvalidOperationException("InstancedBinding does not contain an observable.");
+                    return target.Bind(property, binding.Observable, binding.Priority);
```

Comment:
Just to be sure, it's expected that we don't read Subject here anymore?

Reply:
Nevermind, Subject will implement IObservable anyway, so it will be handled here.